bindings/node.js/kzg.cxx

```diff
@@ -0,0 +1,366 @@
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sstream>  // std::ostringstream
+#include <algorithm> // std::copy
+#include <iterator> // std::ostream_iterator
+#include <napi.h>
+#include "c_kzg_4844.h"
+#include "blst.h"
+
+Napi::Value throw_invalid_arguments_count(
+  const uint expected,
+  const uint actual,
+  const Napi::Env env
+) {
+  Napi::RangeError::New(
+    env,
+    "Wrong number of arguments. Expected: "
+    + std::to_string(expected)
+    + ", received " + std::to_string(actual)
+  ).ThrowAsJavaScriptException();
+
+  return env.Null();
+}
+
+Napi::Value throw_invalid_argument_type(const Napi::Env env, std::string name, std::string expectedType) {
+  Napi::TypeError::New(
+    env,
+    "Invalid parameter type: " + name + ". Expected " + expectedType
+  ).ThrowAsJavaScriptException();
+
+  return env.Null();
+}
+
+Napi::TypedArrayOf<uint8_t> napi_typed_array_from_bytes(uint8_t* array, size_t length, Napi::Env env) {
+  // Create std::vector<uint8_t> out of array.
+  // We allocate it on the heap to allow wrapping it up into ArrayBuffer.
+  std::unique_ptr<std::vector<uint8_t>> vector =
+      std::make_unique<std::vector<uint8_t>>(length, 0);
+
+  for (size_t i = 0; i < length; ++i) {
+    (*vector)[i] = array[i];
+  }
+
+  // Wrap up the std::vector into the ArrayBuffer.
+  Napi::ArrayBuffer buffer = Napi::ArrayBuffer::New(
+      env,
+      vector->data(),
+      length /* size in bytes */,
```

Comment:
might not matter, but should this be set to the vector capacity rather than length? I dunno of any stdlib that uses a different capacity at construction tho.

Reply:
I see what you mean and I think it should be the byte length supplied to the function, not the allocation for the vector, even in the case where they differ